### PR TITLE
Add activesocket API, return proper Unix.file_descr

### PR DESCRIPTION
**Pull request by toots**
This PR adds the active socket API, which deprecates and replaces the last socket API: https://curl.se/libcurl/c/CURLINFO_ACTIVESOCKET.html

It also changes the return type to return proper `Unix.file_descr`.

I haven't included the autoconf/automake changes since I am not sure how you usually re-generates those.

### CODE REVIEW

File: curl.ml [hunk 1]
```diff
@@ -1162,8 +1164,13 @@ let get_cookielist conn =
 
 let get_lastsocket conn =
   match (getinfo conn CURLINFO_LASTSOCKET) with
-  | CURLINFO_Long l -> l
-  | _ -> 0
+  | CURLINFO_Socket s -> Some s
```
Comment (ygrek) [inline curl.ml:1167]: lets not break compatibility, new code will use activesocket with proper type

**Review by ygrek — CHANGES_REQUESTED** [review]
autoreconf should generate necessary configure changes that can be committed

**Comment by toots** [comment]
Sounds good. I've pushed what should be the final PR. I also fixed the `omulti` example & removed `Libevent` to avoid external dependencies so it can be built by default.

**Comment by ygrek** [comment]
> I also fixed the omulti example & removed Libevent to avoid external dependencies so it can be built by default.

I appreciate the change but I cannot appreciate that it also includes the reformatting of the whole file so it is impossible to review the essential change

**Comment by ygrek** [comment]
actually thanks it is in separate commit

File: examples/omulti.ml [hunk 2]
```diff
@@ -16,73 +16,67 @@ let finished mt =
...
+      let timeout =
+        if !in_fd = [] && !out_fd = [] then 0.1 else -1.
+      in
+      let (fdi,fdo,_) = Unix.select !in_fd !out_fd [] timeout in
+      List.iter (fun fd -> on_event fd M.EV_IN) fdi;
+      List.iter (fun fd -> on_event fd M.EV_OUT) fdo;
+      process ()
+    )
+  in
+  let th = Thread.create process () in
```
Comment (ygrek) [inline examples/omulti.ml:74]: thread is not needed?

Comment (ygrek) [reply examples/omulti.ml:74]: and all the mutexes either? all the callbacks are invoked in the context of select, so it is always same thread, there is no concurrent code running

Comment (ygrek) [reply examples/omulti.ml:74]: actually only now I understand that adding this thread stuff is the only essential change in omulti.ml? Please don't change formatting.

**Review by ygrek — CHANGES_REQUESTED** [review]
also I want to keep libevent example, because in one form or another it is a guidance for those who want to get high performance. Maybe should use conditional compilation, but this is separate change.

**Comment by ygrek** [comment]
Cherry-picked without omulti changes : bc339867711abcbe8b83f51cf27d254eab5616be
Thanks!

**Pull request closed without merging**